figures/scripts/transcriptomic-overview.R

```diff
@@ -0,0 +1,336 @@
+# same order
+short_histology_df <- histologies_df %>%
+  filter(experimental_strategy == "RNA-Seq",
+                RNA_library == "stranded") %>%
+  select(Kids_First_Biospecimen_ID,
+                short_histology) %>%
+  arrange(Kids_First_Biospecimen_ID) %>%
+  tibble::column_to_rownames("Kids_First_Biospecimen_ID") %>%
+  as.data.frame()
+
+gsva_scores_mat <- gsva_scores_mat[, order(colnames(gsva_scores_mat))]
+
+# The colors supplied to HeatmapAnnotation need to be a named
+# list of named vectors
+annotation_colors <- list(
+  short_histology = tibble::deframe(histology_palette)
+)
+
+# Annotation bar intended for the top of the heatmap
+column_heatmap_annotation <- HeatmapAnnotation(
+  df = short_histology_df,
+  col = annotation_colors,
+  name = "short histology",
+  show_legend = FALSE
+)
+
+## Heatmap itself!
+gsva_heatmap <- Heatmap(
+  gsva_scores_mat,
+  col = gsva_col_fun,
+  name = "GSVA Scores",
+  na_col = na_color$hex_codes,
+  show_column_names = FALSE,
+  row_names_gp = grid::gpar(fontsize = 5),
+  top_annotation = column_heatmap_annotation,
+  heatmap_legend_param = list(direction = "horizontal")
+)
+
+png(gsva_png, width = 5.5, height = 5, units = "in", res = 1200)
+draw(gsva_heatmap, heatmap_legend_side = "bottom")
+dev.off()
+
+#### Immune deconvolution ------------------------------------------------------
+
+immune_deconv_dir <- file.path(
+  analyses_dir,
+  "immune-deconv"
+)
+
+immune_deconv_file <- file.path(
+  immune_deconv_dir,
+  "results",
+  "deconv-output-for-figures.RData"
+)
+
+load(immune_deconv_file)
+
+# Get in wide format for making a heatmap
+deconv_mat <- deconv.res %>%
+  select(cell_type, `sample`, fraction) %>%
+  spread(`sample`, fraction) %>%
+  tibble::column_to_rownames("cell_type") %>%
+  as.matrix()
+
+# We only want the stranded data - this particular module combined the two
+# RNA-seq datasets
+deconv_mat <- deconv_mat[, colnames(gsva_scores_mat)]
+
+# We need a new color palette, etc.
+na_color <- gradient_palette %>%
+  filter(color_names == "na_color")
+
+gradient_palette  <- gradient_palette %>%
+  filter(color_names != "na_color")
+
+gradient_col_val <- seq(from = min(deconv_mat),
+                         to = max(deconv_mat),
+                         length.out = nrow(gradient_palette))
+
+deconv_col_fun <- circlize::colorRamp2(gradient_col_val,
+                                       gradient_palette$hex_codes)
+
+# There are 3 general scores from xCell - immune score, microenvironment score,
+# and stroma score
+# We're going to separate those rows out from the cell type scores
+scores_deconv_mat <- deconv_mat[grep("score", rownames(deconv_mat)), ]
+cell_type_deconv_mat <- deconv_mat[-grep("score", rownames(deconv_mat)), ]
+
+# To keep ordering consistent across the two xCell heatmaps, we'll order by
+# histology
+short_histology_df <- short_histology_df %>%
+  # We lose the rownames if we use arrange without this step
+  tibble::rownames_to_column("biospecimen") %>%
```

Comment:
If you arrrange this before we use `short_histology_df` for the first time, can you avoid this bit and then just use `rownames(short_histology_df)`. Or is there a danger that this can get out of order  between that heatmap and this one?